worker_pool.go

```diff
@@ -0,0 +1,48 @@
+package wavelet
+
+import "sync"
+
+type WorkerPool struct {
+	bus    chan func()
+	stop   chan struct{}
+	wg     sync.WaitGroup
+	closed bool
+}
+
+func NewWorkerPool() *WorkerPool {
+	return &WorkerPool{
+		stop: make(chan struct{}),
+		bus:  make(chan func()),
+	}
+}
+
+func (wp *WorkerPool) Start(workers int) {
+	wp.wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go func() {
+			defer wp.wg.Done()
+			for {
+				select {
+				case job := <-wp.bus:
+					job()
+				case <-wp.stop:
+					return
+				}
+			}
+		}()
+	}
+}
+
+func (wp *WorkerPool) Queue(job func()) {
+	wp.bus <- job
+}
+
+func (wp *WorkerPool) Stop() {
```

Comment:
Should it be possible to call `wp.Start()` again after `wp.Stop()` has been called -- or should it just be an error to do so ?

Reply:
no restarts intended. worker pool lifecycle expected to be same as application using it.

Reply:
Excellent, closing `wp.bus` should solve this as well